contracts/Main.sol

```diff
@@ -51,9 +48,7 @@ contract DimensionX is ERC3525SlotEnumerableUpgradeable, Vault, InitLock {
         uint burnTokenBalance = this.balanceOf(fromTokenId_);
         uint getToTokenAmount = (fromSlot * amount_) / slot_;
 
-        uint newTokenId = _tokenIds.current();
-        _mint(msg.sender, newTokenId, slot_, getToTokenAmount);
-        _tokenIds.increment();
+        uint256 newTokenId = _mint(msg.sender, slot_, getToTokenAmount);
```

Comment:
To add multichain support after merging in changes from `main` it should be:

uint256 newTokenId = _mint(spanningMsgSender(), slot_, getToTokenAmount);

Reply:
thanks for your suggestion，I will update this